source/non_matching/fe_values.cc

```diff
@@ -0,0 +1,286 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2021 - 2021 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of deal.II.
+//
+// ---------------------------------------------------------------------
+
+#include <deal.II/base/exceptions.h>
+
+#include <deal.II/lac/block_vector.h>
+#include <deal.II/lac/la_parallel_block_vector.h>
+#include <deal.II/lac/la_parallel_vector.h>
+#include <deal.II/lac/la_vector.h>
+#include <deal.II/lac/petsc_block_vector.h>
+#include <deal.II/lac/petsc_vector.h>
+#include <deal.II/lac/trilinos_epetra_vector.h>
+#include <deal.II/lac/trilinos_parallel_block_vector.h>
+#include <deal.II/lac/trilinos_tpetra_vector.h>
+#include <deal.II/lac/trilinos_vector.h>
+#include <deal.II/lac/vector.h>
+
+#include <deal.II/non_matching/fe_values.h>
+
+#include <deal.II/numerics/fe_field_function.h>
+
+DEAL_II_NAMESPACE_OPEN
+
+namespace NonMatching
+{
+  RegionUpdateFlags::RegionUpdateFlags()
+    : inside(update_default)
+    , outside(update_default)
+    , surface(update_default)
+  {}
+
+
+  namespace internal
+  {
+    namespace FEValuesImplementation
+    {
+      template <int dim, class VECTOR>
+      class RefSpaceFEFieldWrapper : public LevelSetDescription<dim>
+      {
+      public:
+        RefSpaceFEFieldWrapper(const DoFHandler<dim> &dof_handler,
+                               const VECTOR &         level_set);
+
+
+        const Function<dim> &
+        get_ref_space_level_set(
+          const typename Triangulation<dim>::active_cell_iterator &cell)
+          override;
+
+      private:
+        Functions::RefSpaceFEFieldFunction<dim, VECTOR> level_set;
```

Comment:
I guess this is the answer to my question in https://github.com/dealii/dealii/pull/12101#pullrequestreview-754364194.